src/openvic-simulation/country/Country.cpp

```diff
@@ -0,0 +1,233 @@
+}
+
+bool CountryManager::add_country(
+	std::string_view identifier,
+	colour_t color,
+	const GraphicalCultureType& graphical_culture,
+	std::vector<CountryParty>&& parties,
+	std::vector<UnitNames>&& unit_names,
+	bool dynamic_tag,
+	Dictionary<colour_t>&& alternative_colours
+) {
+	if (identifier.empty()) {
+		return false;
+	}
+
+	return countries.add_item({ identifier, color, graphical_culture, std::move(parties), std::move(unit_names), dynamic_tag, std::move(alternative_colours) });
+}
+
+bool CountryManager::load_country_data_file(GameManager& game_manager, std::string_view name, bool is_dynamic, ast::NodeCPtr root) {
+	colour_t color;
+	const GraphicalCultureType* graphical_culture;
+	std::vector<CountryParty> country_parties;
+	std::vector<UnitNames> unit_names;
+
+	bool ret = expect_dictionary_keys(
+		ALLOW_OTHER_KEYS,
+		"color", ONE_EXACTLY, expect_colour(assign_variable_callback(color)),
+		"graphical_culture", ONE_EXACTLY, expect_identifier_or_string([&game_manager, &graphical_culture, &name](std::string_view value) -> bool {
+			graphical_culture = game_manager.get_pop_manager().get_culture_manager().get_graphical_culture_type_by_identifier(value);
+			if (graphical_culture == nullptr) {
+				Logger::error("When loading country ", name, ", specified graphical culture ", value, " is invalid!\nCheck that CultureManager has loaded before CountryManager.");
+			}
+
+			return graphical_culture != nullptr;
+		}),
+		"party", ZERO_OR_MORE, [&game_manager, &country_parties, &name](ast::NodeCPtr value) -> bool {
+			std::string_view party_name, ideology_name,
+				economic_policy,
+				trade_policy,
+				religious_policy,
+				citizenship_policy,
+				war_policy;
+			Date start_date, end_date;
+
+			bool ret = expect_dictionary_keys(
+				"name", ONE_EXACTLY, expect_identifier_or_string(assign_variable_callback(party_name)),
+				"start_date", ONE_EXACTLY, expect_date(assign_variable_callback(start_date)),
+				"end_date", ONE_EXACTLY, expect_date(assign_variable_callback(end_date)),
+				"ideology", ONE_EXACTLY, expect_identifier(assign_variable_callback(ideology_name)),
+				"economic_policy", ONE_EXACTLY, expect_identifier(assign_variable_callback(economic_policy)),
+				"trade_policy", ONE_EXACTLY, expect_identifier(assign_variable_callback(trade_policy)),
+				"religious_policy", ONE_EXACTLY, expect_identifier(assign_variable_callback(religious_policy)),
+				"citizenship_policy", ONE_EXACTLY, expect_identifier(assign_variable_callback(citizenship_policy)),
+				"war_policy", ONE_EXACTLY, expect_identifier(assign_variable_callback(war_policy))
+			)(value);
+
+			const Ideology* ideology = game_manager.get_ideology_manager().get_ideology_by_identifier(ideology_name);
+			if (ideology == nullptr) {
+				Logger::error("When loading party ", party_name, " of ", name, ", specified ideology ", ideology_name, " is invalid!\nCheck that IdeologyManager has loaded before CountryManager.");
+				return false;
+			}
```

Comment:
`IdeologyManager::expect_ideology_identifier` + `assign_variable_callback_pointer` can be used to parse the ideology into an `Ideology const*`, hiding the intermediate string/identifier form.